netbox_onboarding/onboard.py

```diff
@@ -198,6 +200,41 @@ def get_platform_object_from_netbox(platform_slug):
 
         return platform
 
+    def check_ip(self):
+        """Method to check if the IP address form field was an IP address.
+
+        If it is a DNS name, attempt to resolve the DNS address and assign the IP address to the
+        name.
+
+        Returns:
+            (bool): True if the IP address is an IP address, or a DNS entry was found and
+                    reassignment of the ot.ip_address was done.
+                    False if unable to find a device IP (error)
+
+        """
+        try:
+            # Assign checked_ip to None for error handling
+            # If successful, this is an IP address and can pass
+            checked_ip = netaddr.IPAddress(self.ot.ip_address)
+            return True
+        # Catch when someone has put in a prefix address, raise an exception
+        except ValueError:
+            raise OnboardException(
+                reason="fail-prefix", message=f"ERROR appears a prefix was entered: {self.ot.ip_address}"
+            )
+        # An AddrFormatError exception means that there is not an IP address in the field, and should continue on
+        except AddrFormatError:
+            try:
+                # Do a lookup of name to get the IP address to connect to
+                checked_ip = socket.gethostbyname(self.ot.ip_address)
+                self.ot.ip_address = checked_ip
+                return True
+            except socket.gaierror:
+                # DNS Lookup has failed, Raise an exception for unable to complete DNS lookup
+                raise OnboardException(
+                    reason="fail-dns", message=f"ERROR failed to complete DNS lookup: {self.ot.ip_address}"
```

Comment:
Since you're adding a couple of new possible failure `reason` values, please also add these values into `choices.OnboardingFailChoices` so that they appear as filter options in the web interface.